### PR TITLE
fix(shell): restructure shell styles

### LINKED ISSUE #32: Modularise `Shell` styles
#### Context

Modularise `Shell` component and sub-components styles  - 

- [ ] `Accordion`
- [ ] `Breadcrumb`
- [ ] `Button`
- [ ] `Header`
- [ ] `IconButton`
- [ ] `Nav`
- [ ] `ProfileImage`
- [ ] `Toolbar`
- [ ] `Tooltip`

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issues

- Resolves #32 

## Proposed changes

- Update `Shell` and sub-components to use correct partials
- Move Carbon body overrides into `platform/body` to prevent override unless the platform styles are required
- Move Carbon type overrides into `globals/type` to prevent override unless the global styles are required
- Add comments for lines to revert

### CODE REVIEW

File: .storybook/config.js [hunk 1]
```diff
@@ -71,6 +71,8 @@ addParameters({
     ),
     storySort: (a, b) =>
       ORDER.indexOf(getCategory(a)) - ORDER.indexOf(getCategory(b)),
+
+    // TODO: `modular-styles` - Revert.
```
Comment (j1mie) [inline .storybook/config.js]: Good thinking 🎖

File: src/components/Header/_mixins.scss [hunk 2]
```diff
@@ -4,10 +4,22 @@
...
+@import '@carbon/type/scss/styles';
+@import 'carbon-components/scss/globals/scss/helper-mixins';
+@import 'carbon-components/scss/globals/scss/layout';
+
+@import '../../globals/motion/index';
+@import '../../globals/namespace/index';
 
 @import '../../platform/body/mixins';
 
+@import '../IconButton/mixins';
```
Comment (j1mie) [inline src/components/Header/_mixins.scss]: This is one issue I couldn't really figure out - we'd talked about keeping imports from other components in the components `index.scss` but this doesn't always seem to work. Thought?

File: src/components/Header/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
+////
+/// Header variables.
+/// @group header
+/// @copyright IBM Security 2019
+////
+
+@import '@carbon/layout/scss/spacing';
+@import 'carbon-components/scss/globals/scss/layout';
```
Comment (j1mie) [inline src/components/Header/_variables.scss]: Hadn't come across a `_variables.scss` partial before. Interesting. Is this something we're using elsewhere?

File: src/components/Toolbar/_mixins.scss [hunk 4]
```diff
@@ -4,10 +4,17 @@
...
+@import '@carbon/themes/scss/mixins';
+@import 'carbon-components/scss/globals/scss/layout';
+
+@import '../../globals/motion/index';
+@import '../../globals/namespace/index';
+@import '../../globals/rtl/index';
 
 @import '../../platform/body/mixins';
-@import '../Header/mixins';
+@import '../Header/variables';
```
Comment (j1mie) [inline src/components/Toolbar/_mixins.scss]: Makes sense now.

File: src/components/_index.scss [hunk 5]
```diff
@@ -5,6 +5,7 @@
 ////
 
 // Application.
+@import 'Breadcrumb/index';
```
Comment (j1mie) [inline src/components/_index.scss]: These were just missing from the component stylesheet? Guessing they were being brought in by other components?

File: src/globals/layout/_index.scss [hunk 6]
```diff
@@ -4,8 +4,6 @@
 /// @copyright IBM Security 2019
 ////
 
-@import '../variables/index';
-
 @import '@carbon/layout/scss/layout';
```
Comment (j1mie) [inline src/globals/layout/_index.scss]: What's the point of this file

Comment (j1mie) [reply src/globals/layout/_index.scss]: 😄

File: src/components/Header/_mixins.scss (hunk 2 above)
Comment (SimonFinney) [reply src/components/Header/_mixins.scss]: In this file, we explicitly require the `IconButton` mixin, but not the styles themselves

File: src/components/_index.scss (hunk 5 above)
Comment (SimonFinney) [reply src/components/_index.scss]: We had it further below, but I rearranged it to the top

File: src/globals/layout/_index.scss (hunk 6 above)
Comment (SimonFinney) [reply src/globals/layout/_index.scss]: This is a deprecation helper for consumers of older layout variables like `$spacing-*`, `get-fixed-size`, etc.

**Pull request merged**